Disable autovacuum by default.

bmdoil opened the pull request:
Set GUC `autovacuum = false` by default.

For now we will disable autovacuum which also disables auto-analyze. In a future commit we will provide a method to independently enable auto-analyze.

ashwinstar commented:
- won't this also turn off auto-ANALYZE? Context commit 

```
commit 58330d728d20b19b28612f9f2730852b71d52d01
Author: Junfeng(Jerome) Yang <jeyang@pivotal.io>
Date:   Thu Nov 12 09:51:16 2020 +0800

    Enable auto-ANALYZE as defalut for the cluster
    
    Enable the auto-ANALYZE for the cluster as default through the
    `autovacuum` GUC.

  <only partial paste of the commit message>
```

- also, do we really need to turn off reloption default if turning off demon guc?

bmdoil commented:
> * won't this also turn off auto-ANALYZE? Context commit
Yes, we will want to enable auto-ANALYZE independently from the `autovacuum` GUC so will need to follow up on that in a future PR. Added more context to thje commit message.
> * also, do we really need to turn off reloption default if turning off demon guc?
No we do not, updated to only disable `autovacuum`

bmdoil commented:
Closing this PR - we are planning to do a more thorough investigation of autovacuum and have decided to leave the GUC enabled for GP7 beta releases.
